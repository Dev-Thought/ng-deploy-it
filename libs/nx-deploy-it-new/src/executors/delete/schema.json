{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "title": "Delete executor",
  "description": "",
  "type": "object",
  "properties": {},
  "required": []
}
