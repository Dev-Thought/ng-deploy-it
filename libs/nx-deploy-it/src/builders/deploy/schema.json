{
  "$schema": "https://json-schema.org/draft-07/schema",
  "$id": "https://json-schema.org/draft-07/schema",
  "title": "NxDeployIt Deploy infrastructure",
  "description": "Deploy the infrastructure for the app",
  "type": "object",
  "properties": {
    "noBuild": {
      "type": "boolean",
      "default": false,
      "description": "Skip build process during deployment"
    },
    "nonInteractive": {
      "type": "boolean",
      "default": false,
      "description": "Changes will apply without user input"
    }
  },
  "required": []
}
