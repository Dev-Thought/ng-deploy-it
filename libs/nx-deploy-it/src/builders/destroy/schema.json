{
  "$schema": "https://json-schema.org/draft-07/schema",
  "$id": "https://json-schema.org/draft-07/schema",
  "title": "NxDeployIt Destroy infrastructure",
  "description": "Destroy the infrastructure for the app",
  "type": "object",
  "properties": {
    "nonInteractive": {
      "type": "boolean",
      "default": false,
      "description": "Changes will apply without user input"
    }
  },
  "required": []
}
