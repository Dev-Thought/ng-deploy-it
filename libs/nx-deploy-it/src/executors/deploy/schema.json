{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "title": "NxDeployIt Deploy infrastructure",
  "description": "Deploy the infrastructure for the app",
  "type": "object",
  "properties": {
    "nonInteractive": {
      "type": "boolean",
      "default": false,
      "description": "Changes will apply without user input"
    }
  },
  "required": []
}
