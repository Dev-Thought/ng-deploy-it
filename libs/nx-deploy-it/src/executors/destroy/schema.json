{
  "$schema": "http://json-schema.org/schema",
  "title": "NxDeployIt Destroy infrastructure",
  "description": "Destroy the infrastructure for the app",
  "type": "object",
  "properties": {
    "nonInteractive": {
      "type": "boolean",
      "default": false,
      "description": "Changes will apply without user input"
    }
  },
  "required": []
}
