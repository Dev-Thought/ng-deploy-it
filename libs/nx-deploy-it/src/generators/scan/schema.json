{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "scan",
  "title": "",
  "type": "object",
  "cli": "nx",
  "properties": {}
}
