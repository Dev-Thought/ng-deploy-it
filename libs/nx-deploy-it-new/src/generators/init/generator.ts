import {
  addProjectConfiguration,
  convertNxGenerator,
  formatFiles,
  generateFiles,
  getWorkspaceLayout,
  names,
  offsetFromRoot,
  Tree,
} from '@nrwl/devkit';
import * as path from 'path';
import { InitGeneratorSchema } from './schema';

interface NormalizedSchema extends InitGeneratorSchema {
  projectName: string;
  projectRoot: string;
  projectDirectory: string;
  parsedTags: string[];
}

export async function initGenerator(host: Tree, options: InitGeneratorSchema) {
  await formatFiles(host);
}

export default initGenerator;
export const scanSchematic = convertNxGenerator(initGenerator);
