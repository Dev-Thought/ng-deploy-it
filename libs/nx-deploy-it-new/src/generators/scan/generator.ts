import { formatFiles, Tree, logger, convertNxGenerator } from '@nrwl/devkit';
import { ApplicationType } from '../../utils/application-type';
import { QUESTIONS } from '../../utils/questions';
import { getApplications } from '../../utils/workspace';
import { ScanGeneratorSchema } from './schema';
import { prompt } from 'enquirer';
import { PROVIDER } from '../../utils/provider';
import initGenerator from '../init/generator';

interface NormalizedSchema extends ScanGeneratorSchema {
  parsedTags: string[];
}

function normalizeOptions(options: ScanGeneratorSchema): NormalizedSchema {
  const parsedTags = options.tags
    ? options.tags.split(',').map((s) => s.trim())
    : [];

  return {
    ...options,
    parsedTags,
  };
}

export async function scanGenerator(host: Tree, options: ScanGeneratorSchema) {
  // const normalizedOptions = normalizeOptions(options);
  const applications = getApplications(host);
  try {
    const questions: any[] = [];

    if (applications.length === 0) {
      logger.log('info', 'no applications found');
      throw new Error();
    }

    logger.log(
      'info',
      `We found ${applications.length} supported applications.`
    );

    const choosenApplications = await prompt<{
      setupApplications: {
        projectName: string;
        applicationType: ApplicationType;
      }[];
    }>({
      ...QUESTIONS.setupApplications,
      choices: applications.map((app) => ({
        name: `${app.projectName} (${app.applicationType})`,
        value: app,
      })),
      result: function (result: string) {
        return Object.values(this.map(result));
      },
    } as any);

    if (choosenApplications.setupApplications.length === 0) {
      logger.log('info', 'No applications selected. Skipping setup');
      throw new Error();
    }

    const { provider } = await prompt<{ provider: PROVIDER }>([
      QUESTIONS.whichProvider,
    ]);

    switch (provider) {
      case PROVIDER.AWS:
        questions.push(QUESTIONS.awsProfile, QUESTIONS.awsRegion);
        break;

      case PROVIDER.AZURE:
        questions.push(QUESTIONS.azureLocation);
        break;

      case PROVIDER.GOOGLE_CLOUD_PLATFORM:
        questions.push(QUESTIONS.gcpProjectId);
        break;

      default:
        break;
    }

    const generatorOptions = await prompt(questions);

    for (const application of applications) {
      await initGenerator(host, {
        provider,
        project: application.projectName,
        ...generatorOptions,
      });
    }
    await formatFiles(host);
  } catch (error) {
    return;
  }
}

export default scanGenerator;
export const scanSchematic = convertNxGenerator(scanGenerator);
