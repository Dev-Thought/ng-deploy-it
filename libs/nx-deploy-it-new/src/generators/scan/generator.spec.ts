import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { addProjectConfiguration, logger, Tree } from '@nrwl/devkit';
import { stdin } from 'mock-stdin';

import {
  answerScanQuestions,
  answerScanQuestionsWithNoApp,
} from '../../utils-test/enquirer.utils';
import generator from './generator';
import { ScanGeneratorSchema } from './schema';
import initGenerator from '../init/generator';

jest.mock('../init/generator');

describe('scan generator', () => {
  let appTree: Tree;
  const options: ScanGeneratorSchema = {};
  const logSpy = jest.spyOn(logger, 'log');

  let io = null;
  beforeAll(() => (io = stdin()));
  afterAll(() => io.restore());

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace();
  });

  afterEach(() => {
    logSpy.mockReset();
    (initGenerator as jest.Mock).mockReset();
  });

  it('should run successfully', async () => {
    await generator(appTree, options);
    expect(logger.log).toMatchSnapshot();
  });

  describe('with applications', () => {
    it('should aboard setup because of no selections', async () => {
      addProjectConfiguration(appTree, 'angular', {
        root: '',
        targets: {
          build: { executor: '@angular-devkit/build-angular:browser' },
        },
      });

      answerScanQuestionsWithNoApp(io);

      await generator(appTree, options);
      expect(logger.log).toMatchSnapshot();
    });

    it('should setup the selected angular application', async () => {
      addProjectConfiguration(appTree, 'angular', {
        root: '',
        targets: {
          build: { executor: '@angular-devkit/build-angular:browser' },
        },
      });

      answerScanQuestions(io, 'eastasia');

      await generator(appTree, options);
      expect(initGenerator).toHaveBeenCalledTimes(1);
      expect(logger.log).toMatchSnapshot();
    });

    it('should setup 2 selected applications', async () => {
      addProjectConfiguration(appTree, 'angular', {
        root: 'angular',
        targets: {
          build: { executor: '@angular-devkit/build-angular:browser' },
        },
      });
      addProjectConfiguration(appTree, 'angularCustomWebpack', {
        root: 'angularCustomWebpack',
        targets: {
          build: { executor: '@angular-builders/custom-webpack:browser' },
        },
      });

      answerScanQuestions(io, 'eastasia');

      await generator(appTree, options);
      expect(initGenerator).toHaveBeenCalledTimes(2);
      expect(logger.log).toMatchSnapshot();
    });
  });
});
