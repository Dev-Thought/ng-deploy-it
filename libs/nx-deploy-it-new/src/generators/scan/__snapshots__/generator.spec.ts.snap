// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scan generator should run successfully 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "info",
      "no applications found",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`scan generator with applications should aboard setup because of no selections 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "info",
      "We found 1 supported applications.",
    ],
    Array [
      "info",
      "No applications selected. Skipping setup",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`scan generator with applications should setup 2 selected applications 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "info",
      "We found 2 supported applications.",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`scan generator with applications should setup the selected angular application 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "info",
      "We found 1 supported applications.",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
